### PR TITLE
Backport 2.x: Updates to the storage format test strategy after starting implementation

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
These are minor updates to the storage format test strategy, clarifying and tweaking a few points based on my experience acquired as I was implementing it.

Needs to go into both 2.x and 3.0.

### CODE REVIEW

**Review by mstarzyk-mobica — APPROVED** [review]

File: docs/architecture/testing/psa-storage-format-testing.md [hunk 1]
```diff
@@ -78,7 +80,9 @@ In particular, the tests must validate that each `PSA_xxx` constant that is stor
 
 In addition, the coverage of key material must ensure that any variation in key representation is detected. See [“Considerations on key material representations”](#Considerations-on-key-material-representations) for considerations regarding key types.
 
-Method: Each test case creates a key with `psa_import_key`, purges it from memory, then reads it back and exercises it. Generate test cases automatically based on an enumeration of available constants and some knowledge of what attributes (sizes, algorithms, …) and content to use for keys of a certain type. Note that the generated test cases will be checked into the repository (generating test cases at runtime would not allow us to test the stability of the format, only that a given version is internally consistent).
+Method: Each test case creates a key with `psa_import_key`, purges it from memory, then reads it back and exercises it.
+
+Generate test cases automatically based on an enumeration of available constants and some knowledge of what attributes (sizes, algorithms, …) and content to use for keys of a certain type.
```
Comment (ronald-cron-arm) [inline docs/architecture/testing/psa-storage-format-testing.md:85]: Referring to the commit message here:
```
Do not require test data to be in the repository
What matters is that we validate that test data is not removed. Keeping the
test data is the most obvious way, but not the only way.
```
What are the other ways you think about? For an old version of the library which we maintain storage backward compatibility with, we need to somehow know the result of the import-and-store operations for a set of key attributes and key material for that old version. Where would that information be if it is not as test data?

Comment (gilles-peskine-arm) [reply docs/architecture/testing/psa-storage-format-testing.md:85]: A test data diff check, similar to the ABI diff check. Work in progress: https://github.com/ARMmbed/mbedtls/pull/4408

Comment (ronald-cron-arm) [reply docs/architecture/testing/psa-storage-format-testing.md:85]: Sorry but I am still a bit lost with this. Thus let's say that we are on version 1.0 of the storage format and we have a PR that introduces changes to the storage format such as for some keys the storage format is not the same anymore and thus we have to increment the storage format version number to let's say 1.1. We have to change the code that reads from the storage such as we are able to read v1.0 and v1.1 keys. How do we check with your script that we do that correctly ?

Comment (gilles-peskine-arm) [reply docs/architecture/testing/psa-storage-format-testing.md:85]: The script isn't meant to handle format version changes. It's meant to handle changes in the test data. If the test data changes in a way that the script can't verify, the change is highlighted for manual review.

Comment (ronald-cron-arm) [reply docs/architecture/testing/psa-storage-format-testing.md:85]: ok I think that the use case I mentioned above is not relevant to this change. Closing this conversation.

**Review by ronald-cron-arm — APPROVED** [review]

**Comment by ronald-cron-arm** [comment]
Oops, it seems I missed the fact that this needs a forward port to development. @gilles-peskine-arm, please do the forward port.

**Pull request merged**